### PR TITLE
Give it some love

### PR DESCRIPTION
**Pull request by erikfrerejean**
This is a work in progress in every sense of the word!

### CODE REVIEW

**Comment by michaelcullum** [comment]
I like new feature that not only does it say the travis build failed below the opening, it also says: 
Merge with caution.

File: blog/composer.json [hunk 1]
```diff
@@ -1,35 +1,39 @@
...
-            "role": "Developer"
-        }
-    ],
-    "require": {
-        "phpbb": "3.1.0"
-    },
-    "extra": {
-        "display-name": "phpBB Blog"
- }
+	"name": "phpBBBlog/blog",
```
Comment (michaelcullum) [inline blog/composer.json]: Needs to be same as repo location (which will eventually be phpbb-blog with github.com/phpbb-blog/phpbb-blog-for-3.1 as a mirror (Pushing the repo to both locations, locking one).

File: blog/composer.json [hunk 2]
```diff
@@ -1,35 +1,39 @@
...
+			"homepage": "http://michaelcullum.com",
+			"role": "Developer"
+		}
+	],
+	"require": {
+		"php": ">=5.3",
+		"phpbb": "3.1.0-dev"
+	},
+	"extra": {
+		"display-name": "phpBB Blog Extension"
```
Comment (michaelcullum) [inline blog/composer.json]: Not sure we need to say extension here.

File: blog/composer.json [hunk 3]
```diff
@@ -1,35 +1,39 @@
...
+		{
+			"name": "Michael Cullum",
+			"username": "Unknown Bliss",
+			"email": "m@michaelcullum.com",
+			"homepage": "http://michaelcullum.com",
+			"role": "Developer"
+		}
+	],
+	"require": {
+		"php": ">=5.3",
```
Comment (michaelcullum) [inline blog/composer.json]: 5.3.3 now

File: blog/composer.json [hunk 4]
```diff
@@ -1,35 +1,39 @@
...
+		{
+			"name": "David King",
+			"username": "imkingdavid",
+			"email": "imkingdavid@gmail.com",
+			"homepage": "http://thedavidking.com",
+			"role": "Developer"
+		},
+		{
+			"name": "Michael Cullum",
+			"username": "Unknown Bliss",
```
Comment (michaelcullum) [inline blog/composer.json]: Seeing as you're in the file could you update this please? Thankies

File: blog/ext.php [hunk 5]
```diff
@@ -1,8 +1,8 @@
 <?php
-
-class phpbb_ext_blog_blog_ext extends phpbb_extension_base
```
Comment (michaelcullum) [inline blog/ext.php:3]: You should still keep this class here.

**Comment by michaelcullum** [comment]
<3 controllers

File: blog/composer.json [hunk 6]
```diff
@@ -1,35 +1,39 @@
...
-    "require": {
-        "phpbb": "3.1.0"
-    },
-    "extra": {
-        "display-name": "phpBB Blog"
- }
+	"name": "phpBBBlog/blog",
+	"type": "phpbb3-extension",
+	"description": "A simple blog extension for phpBB Ascraeus.",
+	"homepage": "n/a",
```
Comment (michaelcullum) [inline blog/composer.json]: Repo. Or maybe make some gh-pages?

File: blog/composer.json [hunk 7]
```diff
@@ -1,35 +1,39 @@
...
-    ],
-    "require": {
-        "phpbb": "3.1.0"
-    },
-    "extra": {
-        "display-name": "phpBB Blog"
- }
+	"name": "phpBBBlog/blog",
+	"type": "phpbb3-extension",
+	"description": "A simple blog extension for phpBB Ascraeus.",
```
Comment (michaelcullum) [inline blog/composer.json]: Ascraeus -> 3.1

File: blog/composer.json (hunk 6 above)
Comment (michaelcullum) [inline blog/composer.json]: Or blank. n/a is not valid

File: blog/ext.php (hunk 5 above)
Comment (erikfrerejean) [inline blog/ext.php:3]: Why? That whole file is only used for lookups, the file itself is never included, let alone that the class is actually called.

File: blog/composer.json (hunk 1 above)
Comment (erikfrerejean) [inline blog/composer.json]: So `phpBB-Blog/phpBB-Blog-for-3.1`? I couldn't find any documentation on this file so kinda guessed it.

File: blog/composer.json (hunk 2 above)
Comment (erikfrerejean) [inline blog/composer.json]: Well it is an extension, but I don't mind removing it.

File: blog/composer.json (hunk 3 above)
Comment (erikfrerejean) [inline blog/composer.json]: Changed

File: blog/composer.json (hunk 4 above)
Comment (erikfrerejean) [inline blog/composer.json]: Yes

File: blog/composer.json (hunk 6 above)
Comment (erikfrerejean) [inline blog/composer.json]: Blanked it for now. I was planning to throw up some gh-pages at some point.

File: blog/ext.php (hunk 5 above)
Comment (michaelcullum) [inline blog/ext.php:3]: Actually it (supposedly) is a class that is called when actions are made and you can add an install(), uninstall() etc. methods which are run when actions are done.

File: blog/composer.json (hunk 1 above)
Comment (michaelcullum) [inline blog/composer.json]: `phpbb-blog/phpbb-blog`
as it will be that at some point in future.

File: blog/ext.php (hunk 5 above)
Comment (erikfrerejean) [inline blog/ext.php:3]: Added, I don't know enough about this ;)

File: blog/composer.json (hunk 1 above)
Comment (erikfrerejean) [inline blog/composer.json]: updated

**Comment by erikfrerejean** [comment]
I'm not sure why the test suite fails on travis :-1:. It runs locally on all my systems

**Comment by imkingdavid** [comment]
Looks like it isn't properly connecting to a database on travis.

@unknownbliss that's not a "new" feature; it's been like that for a while. :p

**Comment by michaelcullum** [comment]
It was working in f226b4452a350a28494f1391599eb10942478eee but since then one of the commits killed it.

**Comment by imkingdavid** [comment]
Maybe this one https://github.com/Leximosi/phpBB-Blog-for-3.1/commit/790670eb8b06a05460a5936e6b896428baba8937 ?
(btw who is Leximosi in the URL?)

**Comment by michaelcullum** [comment]
@imkingdavid look at the email: https://github.com/Leximosi ;)

**Comment by erikfrerejean** [comment]
@imkingdavid, it would be strange if that commit killed it as that file never gets called on travis _(or at least shouldn't)_. Need to look at that closer though.

**Comment by erikfrerejean** [comment]
Apparently it is caused by an update in phpBB. The tests work now I've reverted the remote to an earlier state.

**Comment by michaelcullum** [comment]
Well, if we aren't running the latest phpBB version then it doesn't have controllers.

**Comment by erikfrerejean** [comment]
That _vendor_ is **only** used for the unit tests, so controllers don't have anything to do with this issue ;). It's something in the testframework that has been changed between the revision of the submodule now and the current develop revision.

**Comment by michaelcullum** [comment]
We'll need it for functional tests.

**Comment by erikfrerejean** [comment]
It needs to be updated at some point anyway ;). But **currently** its not the problem ;). That is what I'm saying.

**Comment by erikfrerejean** [comment]
Not the prettiest solution, but the tests pass again. Should investigate why the tests fail with that other error flag.

**Comment by erikfrerejean** [comment]
@unknownbliss, @imkingdavid please review this so far. Should be pretty much in the same state as what is currently in `master`, however fully as a proper extension now.

**Comment by michaelcullum** [comment]
Does it work?

**Comment by erikfrerejean** [comment]
I've just updated the extension so that all dependencies _(acp module/permissions/db tables)_ are installed when the user enables the module through the ACP.

**Comment by erikfrerejean** [comment]
@unknownbliss, @imkingdavid. Bump.

**Comment by imkingdavid** [comment]
Sorry, I've been quite busy with school and personal life lately. I'll try to take a look at this soon.

**Comment by michaelcullum** [comment]
Same but with school, phpBB & Lighting stuff.

**Comment by michaelcullum** [comment]
I'm going to merge this. At a later date i'm going to go through and update stuff to be namespaces etc. and i'll review everything then. phpBB Blog Extension is back in business. @erikfrerejean do you want to continue working on it?

**Pull request merged**